### PR TITLE
Add .gitignore content in advanced section 

### PR DESCRIPTION
**Pull request by rwnfoo**
#240 

_Note_: Branched off of `pat-content` hence why it includes PAT stuff

## How to view files:
```
git checkout rwnfoo/gitignore
docsify serve docs
```

### CODE REVIEW

File: docs/23_gitignore.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+## Configure Git to ignore files
+
+You can add a text file in your repository titled `.gitignore` that instructs Git to stop tracking changes of certain files. This also means that these files will not be pushed to GitHub from your local machine.
```
Comment (beardofedu) [inline docs/23_gitignore.md:3]:
```suggestion
While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
```

File: docs/23_gitignore.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+These templates are also available when you create a new repository:
+
+![create-gitignore-repo](img/gitignore-create-repo.png ':size=60%')
+
+#### gitignore.io
+
+[This website](https://www.toptal.com/developers/gitignore) is another resource for you to find appropriate `.gitignore` templates when your project contains multiple programming languages.
+
+![gitignore.io](img/gitignore.io.png)
```
Comment (beardofedu) [inline docs/23_gitignore.md:44]: I don't know if we need this image necessarily, while the other images provide context regarding the use of GitHub provided `.gitignore` solutions, we don't have much insight as to any upcoming changes to this site so we might not be aware of any UI changes coming down the pipe. Including / Excluding this image from the file isn't a blocker, just my $.02

File: docs/23_gitignore.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+1. Log files, e.g. `*.log`
+1. Build files, e.g. `Build/*`
+
+### Here are some ways you can add a `.gitignore` file:
+
+#### Ignore files  locally
+
+In your Terminal or Git Bash, `cd` to the repository of interest and create a `.gitignore` file
+```
+touch .gitignore
```
Comment (beardofedu) [inline docs/23_gitignore.md:17]: In general can we try to be more consistent when presenting the reader with steps to complete a task? In this specific instance, the user has created a `.gitignore` file, but hasn't committed it, added anything to the `.gitignore` file, etc. 

Ideally we would have some exercise that lets us show them how the `.gitignore` file works both from a "I can see how I could potentially use this" perspective and a "Oh, I can see it in action through this activity" perspective. 

Overall, great addition for this topic though 🎉

File: docs/23_gitignore.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+## Configure Git to ignore files
+
+While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
```
Comment (parkerbxyz) [inline docs/23_gitignore.md:3]:
```suggestion
While working within your repository, you might notice system or configuration files created by your operating system or text editor. These are typically files that you don't want Git to track or include in the history of your project. You can configure Git to ignore specific files, files types, and folders by defining them in a text file titled `.gitignore` in your repository's root directory. 
```

File: docs/23_gitignore.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+### Here are some ways you can add a `.gitignore` file:
+
+#### Ignore files  locally
+
+In your Terminal or Git Bash, `cd` to the repository of interest and create a `.gitignore` file
+```
+touch .gitignore
+```
+
```
Comment (parkerbxyz) [inline docs/23_gitignore.md:19]:
```suggestion

To share the ignore rules with other users who clone the repository, commit the `.gitignore` file in to your repository.

```

File: docs/23_gitignore.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+**Files that you might want Git to ignore:**
+1. Secrets, e.g. `*.pem`, `*.key`
+1. Binary files, e.g. `*.pdf*`, `*.docx`
+1. Log files, e.g. `*.log`
+1. Build files, e.g. `Build/*`
+
+### Here are some ways you can add a `.gitignore` file:
+
+#### Ignore files  locally
```
Comment (parkerbxyz) [inline docs/23_gitignore.md:13]:
```suggestion
#### Ignore files for a single repository
```

File: docs/23_gitignore.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+## Configure Git to ignore files
+
+While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
+
+**Files that you might want Git to ignore:**
+1. Secrets, e.g. `*.pem`, `*.key`
+1. Binary files, e.g. `*.pdf*`, `*.docx`
+1. Log files, e.g. `*.log`
+1. Build files, e.g. `Build/*`
```
Comment (parkerbxyz) [inline docs/23_gitignore.md:9]: Maybe this section should be moved down below the steps to create an empty `.gitignore` file.

File: docs/23_gitignore.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
+## Configure Git to ignore files
+
+While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
+
+**Files that you might want Git to ignore:**
+
+1. Secrets, e.g. `*.pem`, `*.key`
+1. Binary files, e.g. `*.pdf*`, `*.docx`
```
Comment (parkerbxyz) [inline docs/23_gitignore.md:8]:
```suggestion
1. Binary files, e.g. `*.pdf`, `*.docx`
```

File: docs/23_gitignore.md (hunk 3 above)
Comment (parkerbxyz) [reply docs/23_gitignore.md:19]: This might be useful here: https://gist.github.com/octocat/9257657

**Comment by rwnfoo** [comment]
@beardofedu @parkerbxyz Thank you for the feedback! I incorporated all of them. Changes I made: 

- Add activity to commit a `.gitignore` file, using [this template](https://gist.github.com/octocat/9257657) 
- Moved `github/gitignore` and `gitignore.io` content up such that the activities are at the bottom of the page, consistent with formats in other pages
- Removed `gitignore.io` image

File: docs/23_gitignore.md [hunk 9]
```diff
@@ -2,47 +2,110 @@
...
+
+### Activity: Creating a `.gitignore` file
 
 #### Ignore files for a single repository
 
-In your Terminal or Git Bash, `cd` to the repository of interest and create a `.gitignore` file
+1. In your Terminal or Git Bash, `cd` to a repository of interest and create a `.gitignore` file:
 
 ```sh
 touch .gitignore
```
Comment (beardofedu) [inline docs/23_gitignore.md:37]: `touch` doesn't work if the user is not using a bash-based terminal. Windows CMD users can use: `echo >> .gitignore` to accomplish the same thing.

File: conflict-template/HEAD [hunk 10]
```diff
@@ -0,0 +1 @@
+ref: refs/heads/master
```
Comment (beardofedu) [inline conflict-template/HEAD:1]: Where did all these `conflict-template/*` files come from? I think they were added by mistake....

Comment (rwnfoo) [reply conflict-template/HEAD:1]: @beardofedu oh shoot, you're right, another reason why `git add . ` is baaad 😞

**Pull request merged**